docs/products/m3db/concepts/scaling-m3.rst

```diff
@@ -0,0 +1,162 @@
+
+In our experience memory has been the hardest to scale in M3 clusters. The memory usage of M3 consists of:
+
+* fixed base cost (pretty low, ~hundreds of megabytes if configured with small or no pools)
+
+* per-request overhead (main cost in M3Coordinator)
+
+* cached data for queries
+
+* storing data being aggregated in memory; only applicable if aggregated namespaces are configured (main cost in M3Aggregator or significant one in M3Coordinator if M3Aggregator is not in use)
+
+* storing data not yet written to disk - current block (main cost in M3DB)
+
+The first three are not that interesting as they do not scale directly with configuration and-or points stored. However, the last two are usually the ones that cause scalability problems.
+
+The data being stored for aggregation scales with number of namespaces, times number of time series, and also number of points fitting in a single block for that namespace. Due to that, each added namespace adds significant memory cost (unless some of the time series are filtered).
+
+The data in current blocks (not yet persisted to disk as filesets) has also same pattern. It grows as any of the following grow:
+
+* number of time series
+
+* number of points within single block
+
+* number of configured namespaces
+
+For efficient on-disk storage, M3's default recommendation is to have about 720 points per block. So for 15 seconds per point case, that would indicate 180 minutes, or 3 hours per block.  Following the recommendation may be worth it for high resolution namespaces, but for longer lived ones it may lead to too much data being kept in memory. For example, if there is a point per day, closing block every two years would not work. In that sort of case, e.g. day(s) or at most week's block size duration is recommended.
+
+The data within current block (and for a bit longer depending on buffer past duration) is stored in RAM until they are flushed to disk, and this is multiplied by number of time series AND it is also multiplied by number of namespaces that contain those particular points. 
+
+Given a consistent amount of points per second coming in, the memory usage can be reduced by:
+
+* having smaller block size duration (disk usage will grow faster)
+
+* having fewer namespaces
+
+* filtering data that gets to aggregated namespaces 
+
+* configuring shorter buffer past duration as data for current block is retained in memory that much longer after the block is already past; Aiven for M3 defaults to maximum value, which is block size - 1 second); note that if changing buffer past duration to e.g. 1 minute, data older than 1 minute will be simply discarded
+
+For M3DB memory usage cost, using the disk example, what is in RAM per series is (at least)::
+
+    (U) 4 * 60 * 2 = 480 points +
+
+    (A) 4 * 60 * 4 = 960 points +
+
+    (B) 60 * 4 = 240 points +
+
+    (C) 12 = 12 points
+
+    =~ 1700 points + whatever the not-quite-insignificant per-time-series overhead is (* replication factor)
```

Comment:
"whatever the not-quite-insignificant per-time-series overhead is (* replication factor)" -> time-series point overhead * replication_factor